File: Python3/Repeated Substring Pattern.py
# 68ms, 58.20%
class Solution:
    def repeatedSubstringPattern(self, s):
        """
        :type s: str
        :rtype: bool
        """
        size = len(s)
        for i in range(1, size // 2 + 1):
            if size % i != 0: continue
            if s == s[:i] * (size // i): return True
        return False
    

File: Cpp/Repeated Substring Pattern.cpp
// 16ms, 98.28%
static int x = []() {
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);
    return 0;
}();

class Solution
{
  public:
    bool repeatedSubstringPattern(const string &s)
    {
        int size = s.size(), limit = size / 2;
        for (int i = 1; i <= limit; ++i) {
            if (size % i != 0) continue;
            int count = size / i;
            string str = "", tmp = s.substr(0, i);
            while (count--) str += tmp;
            if (str == s) return true;
        }
        return false;
    }
};
